recipes/qwt/all/conanfile.py

```diff
@@ -86,12 +89,9 @@ def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def generate(self):
+        VirtualBuildEnv(self).generate()
         if self._is_legacy_one_profile:
-            env = VirtualRunEnv(self)
-            env.generate(scope="build")
-        else:
-            env = VirtualBuildEnv(self)
-            env.generate()
+            VirtualRunEnv(self).generate(scope="build")
```

Comment:
You need to include `VirtualRunEnv(self).generate(scope="build")` without the if condition for moc to work correctly in the Conan 2 build as well.

Reply:
Thanks for the feedback @valgur